src/main/java/frc/robot/subsystems/Turret.java

```diff
@@ -47,16 +49,52 @@ public void configure() {
 
     turretMotor.getConfigurator().apply(turretConfig);
   }
+  // "Set" Methods
+
+  /**
+   * Sets the angle of the turret
+   * 
+   * @param angle The angle to set the turret to. <b> Units: </b> Degrees
+   */
+  public void setTurretAngle(double angle) {
+    turretMotor.setControl(positionRequest.withPosition(Units.degreesToRotations(angle)));
+  }
+
+  /**
+   * Reset the turret encoder motor to absolute encoder's value
+   */
+  public void resetTurretToAbsolutePosition() {
+    turretMotor.setPosition(getAbsoluteEncoder());
+  }
 
   /**
-   * Sets the position of the turret
+   * Toggle if the turret should lock onto the position of the speaker. If true,
+   * this will also disable the lock to the amp.
    * 
-   * @param position The position to set the turret to. <b> Units: </b> Degrees
+   * @param lockSpeaker If we should lock on to the speaker
    */
-  public void setTurretPosition(double position) {
-    turretMotor.setControl(positionRequest.withPosition(Units.degreesToRotations(position)));
+  public void setLockSpeaker(boolean lockSpeaker) {
+    if (lockSpeaker == true) {
+      lockAmp = false;
+    }
+    this.lockSpeaker = lockSpeaker;
   }
 
+  /**
+   * Toggle if the turret should lock onto the position of the amp. If true,
+   * this will also disable the lock to the speaker.
+   * 
+   * @param lockSpeaker If we should lock on to the amp
+   */
+  public void setLockAmp(boolean lockAmp) {
+    if (lockAmp == true) {
+      lockSpeaker = false;
+    }
+    this.lockAmp = lockAmp;
+  }
```

Comment:
This may be better as an enum instead of two booleans. But this should work fine